pcmaudiotrack.go

```diff
@@ -0,0 +1,192 @@
+package lksdk
+
+import (
+	"errors"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/gammazero/deque"
+	audio "github.com/livekit/mediatransportutil/pkg/audio"
+	opus "github.com/livekit/mediatransportutil/pkg/audio/opus"
+	rtp "github.com/livekit/mediatransportutil/pkg/audio/rtp"
+	protoLogger "github.com/livekit/protocol/logger"
+	"github.com/pion/webrtc/v4"
+	"go.uber.org/atomic"
+)
+
+const (
+	opusSampleRate = 48000
+)
+
+type PCM16ToOpusAudioTrack struct {
+	*webrtc.TrackLocalStaticSample
+
+	opusWriter         audio.WriteCloser[opus.Sample]
+	pcmWriter          audio.WriteCloser[audio.PCM16Sample]
+	resampledPCMWriter audio.WriteCloser[audio.PCM16Sample]
+
+	frameDuration time.Duration
+
+	sampleBuffer *deque.Deque[audio.PCM16Sample]
+	started      sync.Once
+	ticker       *time.Ticker
+
+	closed atomic.Bool
+	mu     sync.Mutex
+	cond   *sync.Cond
+}
+
+// TODO: Support stereo
+func NewPCM16ToOpusAudioTrack(sampleRate int, frameDuration time.Duration, logger protoLogger.Logger) (*PCM16ToOpusAudioTrack, error) {
+	track, err := webrtc.NewTrackLocalStaticSample(webrtc.RTPCodecCapability{MimeType: webrtc.MimeTypeOpus}, "test", "test")
+	if err != nil {
+		return nil, err
+	}
+
+	opusWriter := audio.FromSampleWriter[opus.Sample](track, opusSampleRate, frameDuration)
+	pcmWriter, err := opus.Encode(opusWriter, 1, logger)
+	if err != nil {
+		return nil, err
```

Comment:
Do we need to close the track here?

Reply:
Do you mean webrtc.TrackRemote? If yes, it does not have a close method, get's cleaned up on OnTrackUnsubscribed or on room.Disconnect()